pkg/template/processor_test.go

```diff
@@ -0,0 +1,223 @@
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"fmt"
+	testtemplates "github.com/codeready-toolchain/member-operator/test/templates"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"context"
+	authv1 "github.com/openshift/api/authorization/v1"
+	projectv1 "github.com/openshift/api/project/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"github.com/satori/go.uuid"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+)
+
+func TestProcess(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		// given
+		scheme := runtime.NewScheme()
+		utilruntime.Must(apitemplate.Install(scheme)) // see https://github.com/openshift/oc/blob/master/cmd/oc/oc.go#L77
+		values := map[string]string{
+			"PROJECT_NAME": "foo",
+			"COMMIT":       "1a2b3c",
+			"USER_NAME":    "developer",
+		}
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, scheme, logger(t))
+
+		// when
+		objs, err := p.Process(templateContent(t), values)
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, objs, 2)
+
+		// project request
+		projectKind := objs[0].Object.GetObjectKind()
+		require.IsType(t, &unstructured.Unstructured{}, projectKind)
+		projectRequest := projectKind.(*unstructured.Unstructured)
+		prJson, err := projectRequest.MarshalJSON()
+		require.NoError(t, err, "failed to marshal json for projectrequest")
+		assert.Equal(t, expectedProjectRequest(), string(prJson))
+
+		// role binding
+		rbKind := objs[1].Object.GetObjectKind()
+		require.IsType(t, &unstructured.Unstructured{}, rbKind)
+		roleBinding := rbKind.(*unstructured.Unstructured)
+		rbJson, err := roleBinding.MarshalJSON()
+		require.NoError(t, err, "failed to marshal json for rolebinding")
+		assert.Equal(t, expectedRoleBinding(), string(rbJson))
+	})
+
+	t.Run("random extra param - fail", func(t *testing.T) {
+
+		// given
+		scheme := runtime.NewScheme()
+		utilruntime.Must(apitemplate.Install(scheme)) // see https://github.com/openshift/oc/blob/master/cmd/oc/oc.go#L77
+
+		random := uuid.NewV4().String()
+		values := map[string]string{
+			"PROJECT_NAME": "foo",
+			"COMMIT":       "1a2b3c",
+			"USER_NAME":    "developer",
+			random:         random,
+		}
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, scheme, logger(t))
+
+		// when
+		objs, err := p.Process(templateContent(t), values)
+
+		// then
+		require.EqualError(t, err, fmt.Sprintf("unknown parameter name \"%s\"", random))
+		require.Len(t, objs, 0)
+	})
+}
+
+func TestProcessAndApply(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		s := scheme.Scheme
+		err := apis.AddToScheme(s)
+		require.NoError(t, err)
+		utilruntime.Must(apitemplate.Install(s)) // see https://github.com/openshift/oc/blob/master/cmd/oc/oc.go#L77
+
+		templateContent := templateContent(t)
+		pn := uuid.NewV4().String()
+		u := uuid.NewV4().String()
+		c := uuid.NewV4().String()
```

Comment:
please use names for the variables. It's fine for one variable that you use one char, but in this combination it's not clear